### PR TITLE
Add examples/extension-demo folder with simplified image extension panel example

### PR DESCRIPTION
**Pull request by esthersweon**
**Public-Facing Changes**

- Add examples/extension-demo folder with simplified image extension panel example
- Will be used as reference in tutorial blog post

### CODE REVIEW

File: examples/extension-demo/src/ExamplePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  useEffect(() => {
+    if (state.topic == undefined) {
+      setState({ topic: imageTopics[0]?.name });
+    }
+  }, [state.topic, imageTopics]);
+
+  // Save our state to the layout when the topic changes.
+  useEffect(() => {
+    context.saveState({ topic: state.topic });
+  }, [state.topic]);
```
Comment (esthersweon) [inline examples/extension-demo/src/ExamplePanel.tsx:67]: @foxymiles - confirming that we need `context` as a dependency here / this `useEffect` can be combined with the one on line 51?

Comment (esthersweon) [reply examples/extension-demo/src/ExamplePanel.tsx:67]: This would apply to the existing image extension panel example as well.

Comment (foxymiles) [reply examples/extension-demo/src/ExamplePanel.tsx:67]: Yes you're right. I think we still need to set up the eslint rules for hooks for this repo. In practice `context` won't change but for correctness we should include it in the dependencies list.

**Review by foxymiles — APPROVED** [review]

File: examples/extension-demo/src/ExamplePanel.tsx (hunk 1 above)
Comment (esthersweon) [reply examples/extension-demo/src/ExamplePanel.tsx]: Pushed follow-up commit; will open separate PR for the existing image extension panel example.

**Pull request merged**